src/main/java/com/ibm/watson/apis/conversation_enhanced/rest/ProxyResource.java

```diff
@@ -128,7 +130,9 @@ private MessageResponse getWatsonResponse(MessageRequest request, String id) thr
 
       // Extract the user's original query from the conversational response
       if (query != null && !query.isEmpty()) {
-        Client retrieveAndRankClient = new Client();
+    	  
+    	  DiscoveryClient discoveryClient = new DiscoveryClient();
```

Comment:
We should have a discoveryClient as class variable instead of creating one every time we get a request